files/KoreBuild/modules/vstest/module.targets

```diff
@@ -111,6 +114,12 @@ Runs the VSTest on all projects in the ProjectToBuild itemgroup.
 
     <PropertyGroup>
       <TestAdapterPath>%(TestAssemblies.RootDir)%(TestAssemblies.Directory)</TestAdapterPath>
+
+      <_TestFilter Condition="'$(RunFlakyTests)' != 'true'">Flaky:All!=true</_TestFilter>
+      <_TestFilter Condition="'$(RunFlakyTests)' == 'true'">Flaky:All=true</_TestFilter>
+
+      <_TestFilter Condition="'$(RunFlakyTests)' != 'true' And '$(AGENT_OS)' != ''">$(_TestFilter)&amp;Flaky:AzP:All!=true&amp;Flaky:AzP:OS:$(AGENT_OS)!=true</_TestFilter>
+      <_TestFilter Condition="'$(RunFlakyTests)' == 'true' And '$(AGENT_OS)' != ''">$(_TestFilter)|Flaky:AzP:All=true|Flaky:AzP:OS:$(AGENT_OS)=true</_TestFilter>
```

Comment:
I'm going to refactor this slightly to ensure that a repo can configure it's own Flaky/Non-Flaky filter logic in `repo.props`, but that won't affect the bulk of this change so please give feedback on that :)

Reply:
This is done now.